sentry_sdk/integrations/bottle.py

```diff
@@ -176,24 +177,32 @@ def size_of_file(self, file):
         return file.content_length
 
 
+def _set_transaction_name_and_source(event, transaction_style, request):
+    # type: (Event, str, Any) -> Event
+    name_for_style = {
```

Comment:
why did you choose this hash style, you now call `transaction_from_function` in some cases even when it's not necessary.

Reply:
Just for readability. I like this hash style. But performance matters, so I changed it. (and it is still readable)